Doberman/BaseMonitor.py

```diff
@@ -130,11 +149,14 @@ def run(self):
         """
         self.logger.debug(f'Starting {self.name}')
         while not self.event.is_set():
-            loop_top = time.time()
-            self.logger.debug(f'Running {self.name}')
-            ret = self.func()
-            if isinstance(ret, (int, float)) and 0. < ret:
-                self.period = ret
+            try:
+                loop_top = time.time()
```

Comment:
This probably needs to be outside the try-except.

Reply:
Yes